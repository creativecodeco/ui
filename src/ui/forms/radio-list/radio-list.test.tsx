import { render } from '@testing-library/react';

import type { RadioItemType, RadioListType } from '@/types';

import RadioList from './radio-list.component';

const baseProps: RadioListType = {
  name: 'test',
  options: Array.from({ length: 4 }).map<RadioItemType>((_, index) => ({
    value: index,
    label: `Option ${index}`
  }))
};

describe('<RadioList />', () => {
  it('snapshot', () => {
    const { container } = render(<RadioList {...baseProps} />);

    expect(container).toMatchSnapshot();
  });

  it('options', () => {
    const { getAllByRole } = render(<RadioList {...baseProps} />);

    expect(getAllByRole('radio').length).toEqual(4);
  });

  it('not options', () => {
    const { container, asFragment } = render(
      <RadioList {...baseProps} options={[]} />
    );

    expect(container).toMatchSnapshot();

    expect(asFragment.length).toEqual(0);
  });

  it('with error', () => {
    const error = 'Error message';
    const { getByText } = render(
      <RadioList {...baseProps} isError error={error} />
    );

    expect(getByText(error)).toBeInTheDocument();
  });
});
