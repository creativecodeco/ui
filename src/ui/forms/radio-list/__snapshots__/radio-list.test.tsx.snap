// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioList /> not options 1`] = `<div />`;

exports[`<RadioList /> snapshot 1`] = `
<div>
  <div
    class="form-control"
  >
    <label
      class="label justify-start gap-3 cursor-pointer"
      for="test"
    >
      <input
        class="radio radio-size-md"
        data-testid="test"
        id="test"
        name="test"
        type="radio"
      />
      <span
        class="label-text"
      >
        Option 0
      </span>
    </label>
    <label
      class="label justify-start gap-3 cursor-pointer"
      for="test"
    >
      <input
        class="radio radio-size-md"
        data-testid="test"
        id="test"
        name="test"
        type="radio"
      />
      <span
        class="label-text"
      >
        Option 1
      </span>
    </label>
    <label
      class="label justify-start gap-3 cursor-pointer"
      for="test"
    >
      <input
        class="radio radio-size-md"
        data-testid="test"
        id="test"
        name="test"
        type="radio"
      />
      <span
        class="label-text"
      >
        Option 2
      </span>
    </label>
    <label
      class="label justify-start gap-3 cursor-pointer"
      for="test"
    >
      <input
        class="radio radio-size-md"
        data-testid="test"
        id="test"
        name="test"
        type="radio"
      />
      <span
        class="label-text"
      >
        Option 3
      </span>
    </label>
  </div>
</div>
`;
